specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2018-02-01-preview/containerInstance.json

```diff
@@ -175,6 +175,12 @@
           },
           "201": {
             "description": "Created - the container group is created.",
+            "headers": {
```

Comment:
I don't see any reason for this header, long running operation are handled by the x-ms-long-running-operation flag

Reply:
When we implemented this feature, we followed the following document about Azure Async Operations:
https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-manager-async-operations

Reply:
I'm sure you implemented the server the right way though :)
I'm saying that inside the Swagger, we have a special flag to say this is a ARM async operation and then generators do all this cumbersome work of headers and polling for free.
So you just use that flag, you define your return type with the final resource, and let Autorest deals with the rest.
Example:
https://github.com/Azure/azure-rest-api-specs/blob/8b34c73ead836efa86806e7da555ade6692cbde4/specification/compute/resource-manager/Microsoft.Compute/stable/2017-12-01/compute.json#L403-L462